docs/core/integration-quality-scale/rules/iqs001.md

```diff
@@ -0,0 +1,70 @@
+---
+title: "IQS001: Config flow"
+---
+
+IQS001 is as follows:
+
+Integration needs to be able to be set up via the UI
+
+## Reasoning
+
+Since its introduction in 2018, the config flow has become the standard way to set up integrations in Home Assistant.
+They allow for a consistent user experience across integrations and provide a way to guide users through the setup process.
+
+Because of the better user experience, we want to make sure that all integrations are able to set up via the config flow.
+
+## Example implementation
+
+To use a config flow in your integration, you need to create a `config_flow.py` file in your integration folder and set `config_flow` in your `manifest.json` to `true`.
+
+`config_flow.py`:
+```python
+"""Config flow for my integration."""
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST
+
+from .const import DOMAIN
+
+
+class MyConfigFlow(ConfigFlow, domain=DOMAIN):
+    """My config flow."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input:
+            return self.async_create_entry(
+                title="MyIntegration",
+                data=user_input,
+            )
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema({vol.Required(CONF_HOST): str}),
+            errors=errors,
+        )
+```
+
+## Additional resources
+
+More information about config flows can be found in the [config flow documentation](../../../config_entries_config_flow_handler).
+
+## Exceptions
+
+There are no exceptions to this rule.
```

Comment:
There are exceptions to this rule?

See the ADR (which could be added as an additional resource as well).

Reply:
I kinda feel double about it, do the exceptions to ADR-0010 belong on the scale?

Reply:
Imho yes; we require bronze level for new integrations. Without that, it would conflict with that ADR.

We can update that ADR, but that is a different proposal.

Reply:
But saying an integration doesn't need a config flow doesn't mean we inherently have to rank it on the scale

Reply:
Yes, but now we exempt integration that can't get a config flow from this rule

Reply:
Yup; if we think the other ADR needs adjustment, that is fine, but that we should discuss and raise separately.